lib/Ast_to_Mir.ml

```diff
@@ -3,51 +3,46 @@
 open Core_kernel
 open Mir
 
-let trans_op (op : Ast.operator) =
-  match op with
-  | Ast.Plus -> Plus
-  | Ast.Minus -> Minus
-  | Ast.Times -> Times
-  | Ast.Divide -> Divide
-  | Ast.Modulo -> Modulo
-  | Ast.Or -> Or
-  | Ast.And -> And
-  | Ast.Equals -> Equals
-  | Ast.NEquals -> NEquals
-  | Ast.Less -> Less
-  | Ast.Leq -> Leq
-  | Ast.Greater -> Greater
-  | Ast.Geq -> Geq
-  | _ ->
-      let msg = " should have been transformed to a FunApp by this point." in
-      raise_s [%message (op : Ast.operator) msg]
+(* XXX fix exn *)
+let unwrap_return_exn = function
+  | Some (Ast.ReturnType ut) -> ut
+  | x ->
+      raise_s [%message "Unexpected return type " (x : Ast.returntype option)]
+
+let rec op_to_funapp op args =
+  { texpr= FunApp (Operators.operator_name op, trans_exprs args)
```

Comment:
One downside of this will be when we want to split up terms in a target expression to build the factor graph, we'll want to match on Plus, but instead we can string match on the name.

Also, just checking, can the user define a "plus" function that might be being shadowed?

Reply:
Yeah, I think that's okay. We can always add a constant string that has the operator names if that helps, or something like that.

> Also, just checking, can the user define a "plus" function that might be being shadowed?

Good catch, that is a problem (well, "Plus"). I guess they need to end with __?

Reply:
> I guess they need to end with __?
I think so? Not sure how that's done.

Reply:
User defined functions cannot shadow any function names in our math library. Semantic checking would reject that. So as long as the function names are chosen such that users would never choose them, eg ending in __, we'd be fine

Reply:
ok, pushed a commit to have them end in __

Reply:
Thanks!